lib/puppet/resource/type.rb

```diff
@@ -360,9 +360,6 @@ def set_resource_parameters(resource, scope)
     resource.add_parameters_from_consume
     inject_external_parameters(resource, scope)
 
-    resource_hash = {}
```

Comment:
I was suggesting adding opportunistically on each method as we learn that it is client side only (like in this case, this particular method).